src/libraries/Microsoft.PowerFx.Core/Utils/StackMarker.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.PowerFx.Core.Utils
+{
+    internal struct StackMarker
```

Comment:
>StackMarker [](http://example.com/codeflow?start=20&length=11)

Can you add a class comment?
I like the abstraction :)
And struct is the way to go (not class) since it's used as a stack parameter.

Reply:
Could we name this StackDepthCounter or anything more meaningful?

Reply:
Renamed to StackDepthCounter.